CMakeLists.txt

```diff
@@ -106,9 +106,14 @@ ENDIF(NOT UNIX)
 # Set option(s)
 option(BUILD_UNITTESTS "Build all unittests and add them as new tests" OFF)
 option(ENABLE_GEANT3_TESTING "Enable tests utilizing Geant3" OFF)
-option(BUILD_MBS "Build MBS" OFF)
 option(BUILD_PROOF_SUPPORT "Support ROOT::Proof (deprecated)" OFF)
 
+option(BUILD_ONLINE "Build the online library" ON)
+option(BUILD_MBS "Build MBS" OFF)
+if (BUILD_MBS)
+  set(BUILD_ONLINE ON)
+endif()
```

Comment:
I think, MBS (at least the part in the MbsAPI directory) can be built without ONLINE? So please try removing this?

Reply:
You can compile MbsAPI without Online for sure but what is the sense of it? You need the MbsAPI only to read some data files which you can't without the Online library. 

So formally you can compile MbsAPI without Online but logically it doesn't make sense.

Reply:
Could you look into [cmake_dependent_option](https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html#command:cmake_dependent_option) and see whether you can use it to model this?

Reply:
In my opinion this will not do what is intended. If I make BUILD_ONLINE depend on BUILD_MBS the option BUILD_ONLINE will be only available if BUILD_MBS is TRUE. In the other case BUILD_MBS is build always since the default value for BUILD_ONLINE is TRUE.

What I want to have is to avoid the case BUILD_MBS=TRUE && BUILD_ONLINE=FALSE since it doesn't make sense from the logical point of view. The code forces that case to BUILD_MBS=TRUE && BUILD_ONLINE=TRUE. All other cases are also valid.

Reply:
> avoid the case BUILD_MBS=TRUE && BUILD_ONLINE=FALSE

What about making MBS depend on ONLINE? That should exactly disallow that case?

Reply:
There a valid cases to build the Online library without MBS support. For example in CBM we need to run FairRunOnline with FairSources which reads our own file format, so we need FairRunOnline but not the MbsAPI.

I would like to get rid of the MbsAPI which we only copied from another project to allow R3B  to read their data. I am not aware of another experiment still using the old MBS data aquisition.

Reply:
> In the other case when BUILD_MBS depends on BUILD_ONLINE the MbsAPI is build always since the default value for BUILD_ONLINE is TRUE. This is something I don't want.

I read the documentation of that CMake feature a bit differently. I think, if MBS depends on ONLINE, then:

* If ONLINE is not enabled, MBS will not be available at all (see your requirement)
* If ONLINE is available, then MBS will be set to `<value>`, which can be `OFF`.

```cmake
cmake_dependent_option(BUILD_MBS "Build MBS" OFF "BUILD_ONLINE" OFF)
```

Reply:
This results in the following 

```
cmake  <sourcedir>
=>
BUILD_MBS:BOOL=OFF
BUILD_ONLINE:BOOL=ON

cmake  <sourcedir> --DBUILD_ONLINE=OFF
=> CMake crash since BUILD_MBS isn't known any longer 
CMake Error at examples/advanced/MbsTutorial/CMakeLists.txt:30 (target_link_libraries):
  Target "ExMbs" links to:

    FairRoot::Online

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```
The crash is understandable since it is checked if BUILD_MBS is TRUE to build some of the tests. Changing the check to

`if(BUILD_MBS AND BUILD_ONLINE)`

solves the issue.

I still don't like the proposed change. First off all I have to use an additional include file and more important I have to read the documentation to understand the new command and its logic. I such still prefer the implemented if statement. 

If you insist on the new function I will change the code.

Reply:
First of all: This was only a suggestion to make the dependencies more "coded" and have CMake do the work.

The crash above confuses me a bit, because BUILD_MBS should be OFF, if BUILD_ONLINE is OFF. Anyway.

As MBS will hopefully go away soon, go ahead with your original idea.

Reply:
The problem is that BUILD_MBS isn't available if BUILD_ONLINE is OFF. The option is only available if the other option is ON.